src/app/dashboard-layout/courses-list/add-course/course-authors/course-authors.component.ts

```diff
@@ -53,53 +88,136 @@ export class CourseAuthorsComponent {
    */
   public isAddOnBlur = true;
 
+  constructor(@Inject(FormBuilder) fb: FormBuilder, authorsFacade: AuthorsFacade) {
+    this.fb = fb;
+    this.onTouched = () => {};
+    this.onChange = (_: any) => {};
+    this.authorsFacade = authorsFacade;
+  }
+
   /**
-   * The list of key codes that will trigger a chipEnd event
+   * ngOnInit
    */
-  public separatorKeysCodes: Array<number> = KEY_KODES;
+  public ngOnInit(): void {
+    this.authorsCourseForm = this.fb.group({
+      authors: [[], [Validators.required]],
+      userInput: [""],
+    });
+
+    this.filteredOptions$ = this.authorsCourseForm.get("userInput").valueChanges.pipe(
+      startWith([]),
+      debounceTime(DEBOUNCE_TIME),
+      switchMap(value => this.filterAuthors(value)),
+    );
+  }
 
   /**
-   * Event emitter for authors changes
+   * This method is called by the forms API to write to the view when
+   * programmatic changes from model to view are requested.
    */
-  @Output()
-  public changeAuthors: EventEmitter<Array<AuthorsModel>> = new EventEmitter();
+  public writeValue(value: Dictionary<Array<AuthorsModel>>): void {
+    const valid: boolean = !isNil(value) && !isNil(value.authors);
+
+    if (valid) {
+      this.authors = value.authors;
+      this.authorsCourseForm.patchValue({ authors: this.authors });
+    }
+  }
+
+  /**
+   * Registers a callback function that is called when the control's value
+   * changes in the UI.
+   */
+  public registerOnChange(fn: any): void {
```

Comment:
This approach is valid, but I think it would be more obvious to output event from component